### PR TITLE
parser:support order by in select from dual

### PR DESCRIPTION
**Pull request by mail2fish**
Thank you for working on TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.

## What have you changed? (mandatory)

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

Just for compatibly the mysql server's behaviour

## What are the type of the changes (mandatory)?

A improvement for the issue:

https://github.com/pingcap/tidb/issues/6782

## How has this PR been tested (mandatory)?

Passed make dev test.

## Does this PR affect documentation (docs/docs-cn) update? (optional)

nope

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -451,6 +451,7 @@ func (s *testParserSuite) TestDMLStmt(c *C) {
 		{"select 1 from dual limit 1", true},
 		{"select 1 where exists (select 2)", false},
 		{"select 1 from dual where not exists (select 2)", true},
+		{"select 1 as a from dual order by a", true},
```
Comment (zz-jason) [inline parser/parser_test.go]: how about:
```sql
select 1 as a from dual order by b;
```

**Comment by mail2fish** [comment]
@shenli  Where  is the run-all-tests?

**Comment by shenli** [comment]
@mail2fish run-all-tests will trigger the internal CI. We will handle it.

**Comment by mail2fish** [comment]
@shenli  Is there something I can do for the jenkins job failed?

**Comment by shenli** [comment]
I rerun the CI to make a double check.

**Comment by mail2fish** [comment]
Got it, I will check it.

**Comment by mail2fish** [comment]
On mysql 5.7

select 1 from dual where 1 < any (select 2);
+---+
| 1 |
+---+
| 1 |
+---+

On Tidb

select 1 from dual where 1 < any (select 2);
+-------------+
| 1 from dual |
+-------------+
|           1 |
+-------------+

**Comment by mail2fish** [comment]
@shenli  It seem All checks have passed.

**Pull request merged**